libs/asset-viewer/src/lib/components/asset-search-detail/asset-search-detail.component.ts

```diff
@@ -1,16 +1,44 @@
+import { HttpClient } from '@angular/common/http';
 import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { map } from 'rxjs';
 
 import { ApiError } from '@asset-sg/client-shared';
 import { ORD } from '@asset-sg/core';
+import { AssetFile } from '@asset-sg/shared';
 
 import { AssetDetailVM } from '../../state/asset-viewer.selectors';
 
 @Component({
-    selector: 'asset-sg-asset-search-detail',
-    templateUrl: './asset-search-detail.component.html',
-    styleUrls: ['./asset-search-detail.component.scss'],
-    changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'asset-sg-asset-search-detail',
+  templateUrl: './asset-search-detail.component.html',
+  styleUrls: ['./asset-search-detail.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class AssetSearchDetailComponent {
-    @Input() public rdAssetDetail$?: ORD.ObservableRemoteData<ApiError, AssetDetailVM>;
+  @Input() public rdAssetDetail$?: ORD.ObservableRemoteData<ApiError, AssetDetailVM>;
+
+  constructor(private httpClient: HttpClient) {
+  }
+
+  downloadFile(file: Omit<AssetFile, 'fileSize'>, isDownload: boolean = true): void {
+    this.httpClient.get(`/api/file/${file.fileId}`, { responseType: 'blob' }).subscribe((blob) => {
+      const url = URL.createObjectURL(blob);
+      const anchor = document.createElement('a');
+
+      anchor.setAttribute('style', 'display: none');
+      anchor.href = url;
+      anchor.rel = 'noopener noreferrer';
+      if (isDownload) {
+        anchor.download = file.fileName;
+      } else {
+        anchor.target = '_blank';
+      }
+      document.body.appendChild(anchor);
```

Comment:
I had a similar task in GB3 just recently, where we had to download a file from a http request with responsType of blob.
WE explicitly decided to not go this way because it bypasses all the security measures put in place by Angular. Instead, we decided to use the FileSaver library for this task: 

[Filesaver](https://www.npmjs.com/package/file-saver)

Here are two articles talking about this in detail I had when I was researching the same topic for GB3:

[Part 1](https://medium.com/@drevets/you-cant-prompt-a-file-download-with-the-content-disposition-header-using-axios-xhr-sorry-56577aa706d6)
[Part 2](https://medium.com/@nerdyman/prompt-a-file-download-with-the-content-disposition-header-using-fetch-and-filesaver-8683faf565d0)


Other than that, it works fine

Reply:
Filesaver does internally the same as I did here with plain JS. It offers - to my knowledge - no security advantages over just downloading the file ourselves.

The topic that both articles talk about is just the `Content-Disposition` header, which tells the browser how to handle a downloaded file. It is not directly related to security, but rather how we would want to user to see and receive a file.

Note that Filesaver does not handle the `Content-Disposition` header either, we would have to do this ourselves, just like we would have to do here. Since we can cover all of our use cases without handling that header, I would argue we can just leave the solution as-is.